source/common/upstream/health_discovery_service.h

```diff
@@ -78,7 +78,7 @@ class HdsCluster : public Cluster, Logger::Loggable<Logger::Id::upstream> {
   std::function<void()> initialization_complete_callback_;
 
   Runtime::Loader& runtime_;
-  const envoy::config::cluster::v3::Cluster& cluster_;
+  const std::unique_ptr<const envoy::config::cluster::v3::Cluster> cluster_config_;
```

Comment:
How do you feel about making this a value instead and relying on protobuf's support for move semantics to avoid a copy? That the code is a bit clearer about the fact that it can never be null.

Reply:
Done, thanks for the reviews!